sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingSymmetricHashJoinHelper.scala

```diff
@@ -219,10 +222,35 @@ object StreamingSymmetricHashJoinHelper extends Logging {
           attributesWithEventWatermark = AttributeSet(otherSideInputAttributes),
           condition,
           eventTimeWatermarkForEviction)
-        val inputAttributeWithWatermark = oneSideInputAttributes.find(_.metadata.contains(delayKey))
-        val expr = watermarkExpression(inputAttributeWithWatermark, stateValueWatermark)
-        expr.map(JoinStateValueWatermarkPredicate.apply _)
 
+        // For example, if the condition is of the form:
+        //    left_time > right_time + INTERVAL 30 MINUTES
+        // Then this extracts left_time and right_time.
+        val attributesInCondition = AttributeSet(
+          condition.get.collect { case a: AttributeReference => a }
+        )
+
+        // Construct an AttributeSet so that we can perform equality between attributes,
+        // which we do in the filter below.
+        val oneSideInputAttributeSet = AttributeSet(oneSideInputAttributes)
+
+        // oneSideInputAttributes could be [left_value, left_time], and we just
+        // want the attribute _in_ the time-interval condition.
+        val oneSideStateWatermarkAttributes = attributesInCondition.filter { a =>
+            oneSideInputAttributeSet.contains(a)
```

Comment:
What ensures this is the event-time attribute? Is this assured to be `left_time` mentioned in the comment?

Reply:
The interesting thing about this PR is that this _won't_ be the event-time attribute. If we only have one known watermarked column, we have to find the other attribute, and that other attribute is the one for which we do state removal. It does not necessarily need to be an event-time attribute.

Reply:
Can you give an example? 

> Is this assured to be left_time mentioned in the comment?

What about this part?